solver/pb/ops.proto

```diff
@@ -211,4 +199,67 @@ message Definition {
 message HostIP {
 	string Host = 1;
 	string IP = 2;
+}
+
+message FileOp {
+	repeated FileAction actions = 2;
+}
+
+message FileAction {
+	int64 input = 1 [(gogoproto.customtype) = "InputIndex", (gogoproto.nullable) = false]; // could be real input or target (target index + max input index)
+	int64 secondaryInput = 2 [(gogoproto.customtype) = "InputIndex", (gogoproto.nullable) = false]; // --//--
+  int64 output = 3 [(gogoproto.customtype) = "OutputIndex", (gogoproto.nullable) = false];
+	oneof action {
+		FileActionCopy copy = 4;
+		FileActionMkFile mkfile = 5;
+		FileActionMkDir mkdir = 6;
+		FileActionRm rm = 7;
+	}
+}
+
+message FileActionCopy {
+	string src = 1;
+	string dest = 2;
+	ChownOpt owner = 4;
+	int32 mode = 5;
+	bool followSymlink = 6;
+	bool dirCopyContents = 7;
+	bool attemptUnpack = 8;
+	bool createDestPath = 9;
+	bool allowWildcard = 10;
+	bool allowEmptyWildcard = 11;
```

Comment:
this is similar to bash's `nullglob` or something else?

Reply:
`filepath.Match`